source/time_tables/tests/route_schedules_test.cpp

```diff
@@ -443,6 +443,66 @@ BOOST_FIXTURE_TEST_CASE(test_get_all_route_stop_times_with_different_vp_and_hour
             vec_dt({"00:50"_t + one_day, "01:05"_t + one_day, "01:15"_t + one_day}));
 }
 
+// We want: (for a end of service at 2h00)
+//      A      B      C
+// S1 23:30  23:50  00:10
+// S2 23:40  00:00  00:20
+// S3 23:50  00:10  00:30
+//
+// Detail in associated PR https://github.com/CanalTP/navitia/pull/1304
+BOOST_AUTO_TEST_CASE(test_route_schedule_with_different_vp_over_midnight) {
+    ed::builder b = {"20151127"};
+    navitia::type::Calendar *c1;
+
+    boost::gregorian::date begin = boost::gregorian::date_from_iso_string("20150101");
+    boost::gregorian::date end = boost::gregorian::date_from_iso_string("20160101");
+
+    b.vj("L", "111111", "", true, "A", "A")
+        ("st1", "23:30"_t)
+        ("st2", "23:40"_t)
+        ("st3", "23:50"_t);
+    b.vj("L", "111111", "", true, "B", "B")
+        ("st1", "23:50"_t)
+        ("st2", "0:00"_t)
+        ("st3", "0:10"_t);
```

Comment:
It should be `("st1", "23:50"_t)("st2", "24:00"_t)("st3", "24:10"_t)`. Maybe you have the problem because your exports don't satisfy the GTFS specs on the stop_times (on which NTFS is based) : https://developers.google.com/transit/gtfs/reference#stop_times_fields